python/pyspark/sql/functions/builtin.py

```diff
@@ -1853,7 +1853,7 @@ def sum_distinct(col: "ColumnOrName") -> Column:
 
 
 @_try_remote_functions
-def listagg(col: "ColumnOrName", delimiter: Optional[Union[Column, str, bytes]] = None) -> Column:
```

Comment:
Let's keep them supporting `Column`. In general, we should keep `Column` in PySpark for all cases if possible.